docs/certificate support proposal.md

```diff
@@ -0,0 +1,229 @@
+   - Container a string which would contain the container (for instance the keyVaultUrl in the case of the enum being KeyVault, or the Store Key in the case of the Windows cert Store [CurrentUser/My], the path in the case of a certificate from its path)
+  - ReferenceOrValue: a string which would contain the secret identifier or value. This would be the name of the certificate in the case of keyvault, the password in the case of a path on disk, the base64 encoded value in the case of Base64 encoded.
+1. Modify the `MicrosoftIdentityOptions` to add the 2 collections of `CertificateDescription` `ClientCertificates` and `EncryptionCertificate` 
+2. Add a new class to load the certificates as a function of their type. We probably want to have a kind of base class (CertificateLoader), and derived classes for each SourceType. We'd start only by the KeyVault one.
+
+Let's start simple, increment based on feedback
+
+Possible design to discuss:
+
+```CSharp
+   /// <summary>
+    /// Source for a certificate.
+    /// </summary>
+    public enum CertificateSource
+    {
+        /// <summary>
+        /// KeyVault
+        /// </summary>
+        KeyVault,
+
+        /// <summary>
+        /// Base 64 encoded directly in the configuration.
+        /// </summary>
+        Base64Encoded,
+
+        /// <summary>
+        /// Local path on disk
+        /// </summary>
+        Path,
+
+        /// <summary>
+        /// From the certificate store, described by its thumprint.
+        /// </summary>
+        StoreWithThumbprint,
+
+        /// <summary>
+        /// From the certificate store, described by its Distinguished name.
+        /// </summary>
+        StoreWithDistinguihedName,
+    }
+
+    /// <summary>
+    /// Description of a certificate.
+    /// </summary>
+    public class CertificateDescription
+    {
+        /// <summary>
+        /// Type of the source of the certificate.
+        /// </summary>
+        public CertificateSource SourceType { get; set; }
+
+        /// <summary>
+        /// Container in which to find the certificate.
+        /// <list type="bullet">
+        /// <item>If <see cref="SourceType"/> equals <see cref="CertificateSource.KeyVault"/>, then
+        /// the container is the KeyVault base URL</item>
+        /// <item>If <see cref="SourceType"/> equals <see cref="CertificateSource.Base64Encoded"/>, then
+        /// this value is not used</item>
+        /// <item>If <see cref="SourceType"/> equals <see cref="CertificateSource.Path"/>, then
+        /// this value is the path on disk where to find the certificate</item>
+        /// <item>If <see cref="SourceType"/> equals <see cref="CertificateSource.StoreWithDistinguihedName"/>,
+        /// or <see cref="CertificateSource.StoreWithThumbprint"/>, then
+        /// this value is the path to the certificate in the cert store, for instance <c>CurrentUser/My</c></item>
+        /// </list>
+        /// </summary>
+        public string Container { get; set; }
```

Comment:
Personally I don't like this "weakly typed" API. Why not use a builder pattern that resembles the json document you described

```csharp
Certificate.LoadFromFile(string path);
Certificate.LoadFromKeyVault(...)
```

Reply:
@bgavrilMS : so you like the initial json? I think it's a bit confusing.